doc/nrf/ug_multi_image.rst

```diff
@@ -129,58 +132,88 @@ For example:
 
 .. code-block:: Kconfig
 
    module=MCUBOOT
    source "${ZEPHYR_NRF_MODULE_DIR}/subsys/partition_manager/Kconfig.template.build_strategy"
 
+.. _ug_multi_image_variables:
 
 Image-specific variables
 ========================
 
 The child image and parent image are executed in different CMake processes and thus have different namespaces.
 Variables in the parent image are not propagated to the child image, with the following exceptions:
 
-* Any variable named *IMAGENAME*\_FOO in a parent image is propagated to the child image named *IMAGENAME* as FOO.
-* Variables that are in the list ``SHARED_MULTI_IMAGE_VARIABLES`` are propagated to all child images.
+* Any variable named ``<IMAGE_NAME>\_FOO`` in a parent image is propagated to the child image named ``<IMAGE_NAME>`` as ``FOO``.
+* Variables that are in the ``SHARED_MULTI_IMAGE_VARIABLES`` list are propagated to all child images.
 
 With these two mechanisms, it is possible to set variables in child images from either parent images or the command line, and it is possible to set variables globally across all images.
-For example, to change the ``CONF_FILE`` variable for the MCUboot image and the parent image, specify the CMake command as follows::
+For example, to change the ``CONF_FILE`` variable for the MCUboot image and the parent image, specify the CMake command as follows:
+
+.. code-block::
 
    cmake -Dmcuboot_CONF_FILE=prj_a.conf -DCONF_FILE=app_prj.conf
 
 You can extend the CMake command that is used to create the child images by adding flags to the CMake variable ``EXTRA_MULTI_IMAGE_CMAKE_ARGS``.
 For example, add ``--trace-expand`` to that variable to output more debug information.
 
+With ``west``, these configuration variables are passed into CMake by using the ``--`` separator:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- -Dmcuboot_CONF_FILE=prj_a.conf -DCONF_FILE=app_prj.conf
+
+Any configuration option for the application or child image can be passed in this way.
+For example, use the following command to build the ``hello_world`` sample using multi-stage bootloaders and configurations specific to each bootloader image:
+
+.. code-block::
+
+   west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- -DCONFIG_SECURE_BOOT=y -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_BUILD_S1_VARIANT=y -Db0_OVERLAY_CONFIG=overlay-minimal-size.conf -Dmcuboot_CONFIG_LOG_MINIMAL=y
+
+The options given in the command do the following:
+
+* Enable the setting for the first-stage immutable ``b0`` bootloader - :option:`CONFIG_SECURE_BOOT`.
+* Enable the setting for the second-stage upgradable MCUBoot - :option:`CONFIG_BOOTLOADER_MCUBOOT`.
+* Produce a pre-signed variant of the MCUBoot image for ``S1`` - :option:`CONFIG_BUILD_S1_VARIANT`.
+* Build the ``b0`` child image with one of its own overlays.
+* Build the MCUBoot child image with minimal logging features - :option:`CONFIG_LOG_MINIMAL`.
```

Comment:
This whole section does not belong in the multi image guide, most of it is already described here:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#adding-a-bootloader-chain-to-your-application

and the additions in this section belongs there as well.

(Seems I forgot to hit `add comment` on this comment)

Reply:
I would agree if the intent of the lines here was also to be a guide of how to build with the bootloader. However, the intent of these lines is for clearly explaining what each configuration option does in relation to their specific image in the build; which is ultimately the purpose of this section of the multi-image docs.

Removing the lines from this PR is perfectly fine with me. I still support them being a part of this page in #3607 though since they add clarity through example.

Of course, we can also change the example to not be so much of a duplicate, but the immutable/upgradable integration is a good example for the concepts of this section.

Reply:
No, this doesn't add clarity, having all of those in a single line makes it less generic.

What we could write would be something like:

> It is possible to provide configuration settings for child images, either as individual settings or through Kconfig fragments.
> Each child image is referenced using its image name, so to set the configuration option `CONFIG_FOO=val` in the child image `bar`, you can use:
> ```
> cmake -Dbar_CONFIG_FOO=val ...
> ```
>
> A Kconfig fragment can be added to the child image default configuration in similar way.
> To add an extra Kconfig fragment `baz.conf` to `bar`, you can use
> ```
> cmake -Dbar_OVERLAY_CONFIG=baz.conf ...
> ```
>
> It is also possible to provide a custom configuration file as replacement for the default Kconfig file for the child image.
> To use the  custom configuration file ``myfile.conf`` when building `bar`, you can use
> ```
> cmake -Dbar_CONF_FILE=myfile.conf ...
> ```
>
> If your application is including multiple child image, then you can combine the above together, that is:
> - Setting `CONFIG_FOO=val` in main application
> - Adding a Kconfig fragment `baz.conf` to `bar` child image, using `-Dbar_OVERLAY_CONFIG=baz.conf`
> - Using `myfile.conf` as configuration for the `quz` child image, using `-Dquz_CONF_FILE=myfile.conf`
>
> will become:
> ```
> cmake -DCONFIG_FOO=val -Dbar_OVERLAY_CONFIG=baz.conf -Dquz_CONF_FILE=myfile.conf ....
> ```

Reply:
@FrancescoSer thoughts about this?